modules/azure/storage_container_nfsv3/main.tf

```diff
@@ -0,0 +1,37 @@
+terraform {
+  required_version = ">=1.3.0"
+
+  required_providers {
+    azurerm = "=3.24.0"
+  }
+
+  backend "azurerm" {}
+}
+
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_storage_data_lake_gen2_filesystem" "filesystem" {
+  name               = var.storage_container_name
+  storage_account_id = var.storage_account_id
+
+  dynamic "ace" {
+    for_each = var.ace_permissions
+    iterator = item
+    content {
+      scope       = item.value.scope
+      type        = item.value.type
+      permissions = item.value.permissions
+    }
+  }
+}
+
+resource "azurerm_storage_data_lake_gen2_path" "directory" {
+  count = var.directories == null ? 0 : length(var.directories)
```

Comment:
Directories is listed as a possible null here but is a required variable, is this correct?

Reply:
Made directories optional